src/workerd/api/memory-cache.c++

```diff
@@ -134,21 +134,35 @@ void SharedMemoryCache::putWhileLocked(ThreadUnsafeData& data,
     kj::Own<CacheValue>&& value,
     kj::Maybe<double> expiration) const {
   size_t valueSize = value->bytes.size();
+
+  auto writeSpan = IoContext::current().makeTraceSpan("memory_cache_write"_kjc);
+  writeSpan.setTag("key"_kjc, kj::str(key));
+  writeSpan.setTag("value_size"_kjc, static_cast<double>(valueSize));
+  writeSpan.setTag("has_expiration"_kjc, expiration != kj::none);
+
   if (valueSize > data.effectiveLimits.maxValueSize) {
     // Silently drop the value. For consistency, also drop the previous value,
     // if one exists, such that a subsequent read() will not return an outdated
     // value. Note that removeIfExistsWhileLocked(key) will update the
     // totalValueSize if necessary, so we don't need to do that here.
+    writeSpan.setTag("write_rejected"_kjc, true);
+    writeSpan.setTag("rejection_reason"_kjc, kj::str("value_too_large"));
```

Comment:
Totally unrelated to this PR, but as an aside it seems like `tracing::Attribute::Value` could probably be extended to support `LiteralStringConst` as well? There appear to be a lot of calls like this throughout the code base that pass in `kj::str("some C string literal")` as the value for a tag.